### DISCUSSION TITLE
rpc error:  x509: certificate is valid for localhost, argocd-server , does not work for amazon load balancer

### DISCUSSION THREAD

**Discussion by ssaleh97** [argoproj/argo-cd#11047] [Q&A]
I am trying to execute this command from my jenkins pipeline to force argocd to sync new changes :
`ARGOCD_SERVER="a11d0dece3e6944f19d2569021730f11-21404.us-east-1.elb.amazonaws.com" 
  APP_NAME="root-app"
  PROJECT="srov4"
  ARGOCD_SERVER=$ARGOCD_SERVER argocd --grpc-web app sync $APP_NAME --force
  ARGOCD_SERVER=$ARGOCD_SERVER argocd --grpc-web app wait $APP_NAME --timeout 600` 
  
  and I get this error:
  level=fatal msg="rpc error: code = Unknown desc = Post [https://a11d0dece3e6944f19d2569021730f11-21404.us-east-1.elb.amazonaws.com:443/application.ApplicationService/Sync](https://a11d0dece3e6944f19d2569021730f11-21404.us-east-1.elb.amazonaws.com/application.ApplicationService/Sync): x509: certificate is valid for localhost, argocd-server, argocd-server.argocd, argocd-server.argocd.svc, argocd-server.argocd.svc.cluster.local, not a11d0dece3e6944f19d2569021730f11-21404.us-east-1.elb.amazonaws.com"  
  **any suggestions guys ?**

**Comment by alexmt** [comment]
Looks like you are using self signed certificate generated by Argo CD.  To solve the problem you might either configure trusted certificate in [argocd-secret](https://github.com/argoproj/argo-cd/blob/309654cece2292fb09959055e8602b9b6dc68077/docs/operator-manual/argocd-secret.yaml#L13) or use `--insecure` to disable cert validation ( this is not sure in prod )
